tools/rvn/CommandLineApp.cs

```diff
@@ -52,6 +63,209 @@ public static int Run(string[] args)
+            }
+
+            parameters.Progress.AddInfo($"ZIP file was successfully added to this location: {parameters.PackageOutputPath}");
+
+            return 0;
+        }
+
+        private static void ExtractSetupInfoObjectFromFile(CreateSetupPackageParameters parameters, out SetupInfo setupInfo)
+        {
+            if (string.IsNullOrEmpty(parameters.SetupJsonPath))
+            {
+                throw new InvalidOperationException("-s|--setup-json-path not provided");
+            }
+
+            if (File.Exists(parameters.SetupJsonPath) == false)
+            {
+                throw new InvalidOperationException($"-s|--setup-json-path path:{parameters.SetupJsonPath} not found");
+            }
+
+            using (StreamReader file = File.OpenText(parameters.SetupJsonPath))
+            {
+                JsonSerializer serializer = new();
+                setupInfo = (SetupInfo)serializer.Deserialize(file, typeof(SetupInfo));
+            }
+        }
+
+        private static void ValidateSetupInfo(SetupInfo setupInfo, CreateSetupPackageParameters parameters)
+        {
+            if (setupInfo.License == null)
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.License)} must be set");
+            }
+
+            if (setupInfo.License.Keys is null || setupInfo.License.Keys.Any() == false)
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.License.Keys)} must be set");
+            }
+
+            if (string.IsNullOrEmpty(setupInfo.License.Id.ToString()))
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.License.Id)} must be set");
+            }
+
+            if (string.IsNullOrEmpty(setupInfo.License.Name))
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.License.Name)} must be set");
+            }
+
+            if (string.IsNullOrEmpty(setupInfo.Email))
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.Email)} must be set");
+            }
+
+            if (string.IsNullOrEmpty(setupInfo.Domain))
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.Domain)} must be set");
+            }
+
+            if (string.IsNullOrEmpty(setupInfo.RootDomain))
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.RootDomain)} must be set");
+            }
+
+            if (setupInfo.NodeSetupInfos is null || setupInfo.NodeSetupInfos.Any() == false)
+            {
+                throw new InvalidOperationException($"{nameof(setupInfo.NodeSetupInfos)} must be set");
+            }
+
+            foreach (var tag in setupInfo.NodeSetupInfos.Keys.Where(tag => IsValidNodeTag(tag) == false))
+            {
+                throw new InvalidOperationException($"{tag} - node tag must contain only capital letters. Maximum length should be up to 4 characters");
```

Comment:
Shouldn't we pass all tags that are not valid in this exception?

Reply:
done